eclipse-theia/theia
# Cancel Tree.refresh when new `root` set

#### What it does


On application startup, these errors are logged by the `scm` package:

![image](https://user-images.githubusercontent.com/62660806/175633412-870ba6a3-bff2-43be-bb30-4c9bc4d8db2f.png)

This PR addresses the error by modifying the logic for handling `set root` on the `TreeImpl` so that it cancels a `refresh` cycle when a new root is set. Previously, `refresh` would continue running and eventually discover that the `root` with which it began no longer corresponded to the current `root` of the tree, producing the error.

#### How to test


1. Start the application.
2. Assert that the errors noted above no longer occur.
3. Trees should continue to work normally: since `refresh` is called every time `set root` runs, the final `refresh` should still run to completion.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
